### PR TITLE
[windows] Use FML_DCHECK in place of C assert

### PR DESCRIPTION
**Pull request by cbracken**
Now that the embedders depend on FML, migrate assertions to FML_DCHECK, which allows for an explanatory log message to be emitted along with the assertion.

The log strings will slightly increase the size of debug binaries, but will not affect release binaries.

No tests since no semantic changes are made.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/flutter_platform_node_delegate_windows.cc [hunk 1]
```diff
@@ -33,14 +35,15 @@ void FlutterPlatformNodeDelegateWindows::Init(std::weak_ptr<OwnerBridge> bridge,
...
 }
 
 // |ui::AXPlatformNodeDelegate|
 gfx::NativeViewAccessible
 FlutterPlatformNodeDelegateWindows::GetNativeViewAccessible() {
-  assert(ax_platform_node_);
-  return ax_platform_node_->GetNativeViewAccessible();
+  FML_DCHECK(ax_platform_node_)
+      << "AXPlatformNode hasn't been created;
+         return ax_platform_node_->GetNativeViewAccessible();
```
Comment (loic-sharma) [inline shell/platform/windows/flutter_platform_node_delegate_windows.cc:46]: This looks weird. Perhaps:

```suggestion
      << "AXPlatformNode hasn't been created";
  return ax_platform_node_->GetNativeViewAccessible();
```

Comment (cbracken) [reply shell/platform/windows/flutter_platform_node_delegate_windows.cc:46]: That's what I get for sending a patch from my Mac! Nice catch!

File: shell/platform/windows/keyboard_key_handler.cc [hunk 2]
```diff
@@ -95,7 +96,7 @@ void KeyboardKeyHandler::ResolvePendingEvent(uint64_t sequence_id,
     }
   }
   // The pending event should always be found.
-  assert(false);
+  FML_LOG(FATAL) << "No unreplied pending events found";
```
Comment (loic-sharma) [inline shell/platform/windows/keyboard_key_handler.cc:99]: Consider something like:

```suggestion
  FML_LOG(FATAL) << "Could not find pending key event " << sequence_id;
```

**Review by loic-sharma — APPROVED** [review]
Very nice clean up! 🚀

**Review by dkwingsmt — APPROVED** [review]
LGTM!

File: shell/platform/windows/keyboard_key_handler.cc (hunk 2 above)
Comment (cbracken) [reply shell/platform/windows/keyboard_key_handler.cc]: Oh nice - much better! Done!

**Pull request merged**